packages/apps/reputation-oracle/server/src/modules/nda/nda.service.spec.ts

```
@@ -0,0 +1,73 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { NDAService } from './nda.service';
+import { UserRepository } from '../user/user.repository';
+import { AuthConfigService } from '../../config/auth-config.service';
+import { NDASignatureDto } from './nda.dto';
+import { NDAError, NDAErrorMessage } from './nda.error';
+import { faker } from '@faker-js/faker/.';
+
+const mockUserRepository = {
+  updateOne: jest.fn(),
+};
+const validNdaUrl = faker.internet.url();
+const mockAuthConfigService = {
+  latestNdaUrl: validNdaUrl,
+};
+
+describe('NDAService', () => {
+  let service: NDAService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        NDAService,
+        { provide: UserRepository, useValue: mockUserRepository },
+        { provide: AuthConfigService, useValue: mockAuthConfigService },
+      ],
+    }).compile();
+
+    service = module.get<NDAService>(NDAService);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('signNDA', () => {
+    const user: any = {
```

Maybe we generate new user before each test? This way we have independent test runs even if we forget to change something

---

In this particular case I think it is better to leave it inside the signNDA module as it is not being modified and we can optimise the code

---

@flopez7 We can leave it inside of `signNDA` `describe` block and have its own inner `beforeEach` to generate new user.
It's not a big overhead to generate new simple small object, but we get independent test input for each run, which is not the case now because `user` is implicitly changed inside of `signNDA` method (it assigns `nda.url`) and you also change it in last test case  `should throw an error if the user has already signed the NDA`